src/async_impl/decoder.rs

```diff
@@ -39,11 +44,11 @@ enum Inner {
 
     /// A `Gzip` decoder will uncompress the gzipped response content before returning it.
     #[cfg(feature = "gzip")]
-    Gzip(GzipDecoder<Peekable<IoStream>>),
+    Gzip(FramedRead<GzipDecoder<StreamReader<Peekable<IoStream>, Bytes>>, BytesCodec>),
 
     /// A `Brotli` decoder will uncompress the brotlied response content before returning it.
     #[cfg(feature = "brotli")]
-    Brotli(BrotliDecoder<Peekable<IoStream>>),
+    Brotli(FramedRead<BrotliDecoder<StreamReader<Peekable<IoStream>, Bytes>>, BytesCodec>),
```

Comment:
async-compression 0.3.7 with support for tokio 1.0 and bytes 1.0 just came out, so this workaround and tokio-util shouldn't be necessary anymore

Reply:
async-compression doesn't directly support `Stream<Bytes>` for `bytes 1.0`, I'm thinking I'll probably deprecate the `stream` feature entirely and users can use this sort of wrapper always.